include/effects/matrix/PatternCyclicCA.h

```diff
@@ -0,0 +1,581 @@
+                    m_refractory = 0.0;
+                }
+                // Re-seed the pattern with the new mode and parameters.
+                SeedCA();
+                // Reset timers to start the new pattern immediately
+                m_patternTimer = 0;
+                m_frameTimer = 0;
+            }
+            return true; // Setting was handled
+        }
+
+
+        // If the setting name doesn't match any in this class, pass it to the base class.
+        return LEDStripEffect::SetSetting(name, value);
+    }
+
+    // Implement SerializeSettingsToJSON to save the effect's settings.
+    bool SerializeSettingsToJSON(JsonObject& jsonObject) override
+    {
+        // Serialize base class settings first.
+        auto jsonDoc = CreateJsonDocument();
+        JsonObject root = jsonDoc.to<JsonObject>();
+        LEDStripEffect::SerializeSettingsToJSON(root); // Base class method
+
+        // Serialize our specific settings.
+        // Using NAME_OF macro for consistency with SetSetting and FillSettingSpecs
+        jsonDoc[NAME_OF(m_numStates)] = m_numStates;
+        jsonDoc[NAME_OF(m_speed)] = m_speed;
+        jsonDoc[NAME_OF(m_lifetime)] = m_lifetime;
+        jsonDoc[NAME_OF(m_excited)] = m_excited;
+        jsonDoc[NAME_OF(m_refractory)] = m_refractory;
+        jsonDoc[NAME_OF(m_threshold)] = m_threshold;
+        // Serialize the enum as a boolean for compatibility with the existing JSON structure
+        jsonDoc[NAME_OF(m_mode)] = (m_mode == CAMode::Cyclic);
+
+        // Merge our settings into the main JSON object.
+        return SetIfNotOverflowed(jsonDoc, jsonObject, __PRETTY_FUNCTION__);
+    }
+#endif
+
+public:
+    // Constructor
+    PatternCyclicCA() : LEDStripEffect(EFFECT_MATRIX_CYCLIC_CA, "Cyclic CA")
+    {
+        // Initialize member variables to default values
+        m_numStates = 24;
+        m_speed = 60;
+        m_lifetime = 10000;
+        m_excited = 0.03;
+        m_refractory = 0.64;
+        m_threshold = 1;
+        m_mode = CAMode::GreenbergHastings; // Default to Greenberg-Hastings mode using enum
+        m_calcNextGen = &PatternCyclicCA::DoGenerationGH; // Set function pointer.
+        m_nextVal = 1;
+        m_frameTimer = 0;
+        m_patternTimer = 0;
+        m_sum = 0;
+        m_lastDrawTime = 0;
+        m_pb1 = nullptr; // Initialize buffer pointers
+        m_pb2 = nullptr;
+    }
+
+    // Constructor for deserialization from JSON
+    PatternCyclicCA(const JsonObjectConst& jsonObject) : LEDStripEffect(jsonObject)
+    {
+        // Deserialize settings from JSON, using default values if keys are not found
+        m_numStates = jsonObject["numStates"] | m_numStates;
+        m_speed = jsonObject["speed"] | m_speed;
+        m_lifetime = jsonObject["lifetime"] | m_lifetime;
+        m_excited = jsonObject["excited"] | m_excited;
+        m_refractory = jsonObject["refractory"] | m_refractory;
+        m_threshold = jsonObject["threshold"] | m_threshold;
```

Comment:
- By project convention, the JSON properties used to persist settings to storage (and hence construct effects from storage) are three-letter acronyms.
- You can use the "fetch from JSON object or use default" syntax if you like, but I'd recommend using the `if (….is<`_`type`_>())` pattern to check if the property in the `jsonObject` is of the correct type, and only use the property value if it is.

Again, for an example of these two points you can refer to the `JsonObjectConst` constructor of `PatternStocks`.